rest/route/middleware.go

```diff
@@ -231,6 +231,38 @@ func (m *TaskHostAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Reque
 	next(rw, r)
 }
 
+type hostAuthMiddleware struct {
+	sc data.Connector
+}
+
+// NewHostAuthMiddleware returns a route middleware that verifies the request's
+// host ID and secret.
+func NewHostAuthMiddleware(sc data.Connector) gimlet.Middleware {
+	return &hostAuthMiddleware{sc: sc}
+}
+
+func (m *hostAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	hostID, ok := gimlet.GetVars(r)["host_id"]
+	if !ok {
+		hostID = r.Header.Get(evergreen.HostHeader)
+		if hostID == "" {
+			gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+				StatusCode: http.StatusUnauthorized,
```

Comment:
Should this be the status? Not `StatusNotFound`?

Reply:
Yes, because host auth requires host ID and secret, which basically act like username/password. Bad request might also be appropriate.

Reply:
Ah ok, that is fine then.